shell/common/animator.cc

```diff
@@ -107,6 +120,46 @@ void Animator::BeginFrame(fml::TimePoint frame_start_time,
     TRACE_FLOW_END("flutter", "PointerEvent", trace_flow_id);
     trace_flow_ids_.pop_front();
   }
+  while (!pointer_data_.empty()) {
+    PointerData pointer_data = pointer_data_.front();
+    switch (pointer_data.change) {
+      case flutter::PointerData::Change::kDown: {
+        PointerDelta delta;
+        delta.physical_x = pointer_data.physical_x;
+        delta.physical_y = pointer_data.physical_y;
+        delta.last_physical_x = pointer_data.physical_x;
+        delta.last_physical_y = pointer_data.physical_y;
+        down_pointers_[pointer_data.device] = delta;
+      } break;
+      case flutter::PointerData::Change::kCancel:
+      case flutter::PointerData::Change::kUp:
+        down_pointers_.erase(pointer_data.device);
+        break;
+      case flutter::PointerData::Change::kMove:
+        down_pointers_[pointer_data.device].physical_x =
+            pointer_data.physical_x;
+        down_pointers_[pointer_data.device].physical_y =
```

Comment:
Grab an iterator and use that instead of doing the map lookup twice?

Reply:
Done